### PR TITLE
Add the stress test framework

### PR DESCRIPTION
**Pull request by hanahmily**
This introduced a stress test framework to generate fixed traffic to verify several behaviors of the banyand server.
 
## Verify the flushing mechanism of tsdb

Most caches of tsdb are based on the badger which limits the memory usage by a fixed number. But the inverted index's memtable gets flushed by an event from the main storage. 

The below screenshot shows the flushing of the main storage.

main storage flushing

The below diagram indicates the inverted indices flushing under the controller of the main storage.
inverted index flushing

## Check encoding algorithm

This diagram shows the compression ratio of a stream under the case that each instance keeps sending the element to the database server. That's why the ratio is such high(about 98%). We will leverage the e2e test to get a more practical result in the next round. This result only indicates that the algorithm is working as expected.

image

This is the gorilla encoding algorithm's result which is similar to the theory result, around 30%

image

Signed-off-by: Gao Hongtao <hanahmily@gmail.com>

### CODE REVIEW

**Comment by hanahmily** [comment]
Once this PR gets merged, we could move to release the 0.1.0

**Comment by wu-sheng** [comment]
A question, is the flush required to improve in the future?
Do we have disk volume cost? Such as how many GB on disk for 10g traces?

**Comment by hanahmily** [comment]
> A question, is the flush required to improve in the future? 

Yep, we will introduce a unified memory controller to manage caches.

> Do we have disk volume cost? Such as how many GB on disk for 10g traces?

No. 0.2.0 will introduce them.

File: banyand/measure/measure.go [hunk 1]
```diff
@@ -31,8 +31,10 @@ import (
 	"github.com/apache/skywalking-banyandb/pkg/timestamp"
 )
 
-// a chunk is 1MB
-const chunkSize = 1 << 20
+const (
+	plainChunkSize = 1 << 20
+	intChunkSize   = 120
```
[the remark this answers is not in the document]
Comment (hanahmily) [reply banyand/measure/measure.go:36]: 120 means metrics data volume of a series in 2 hours.

File: pkg/logger/setting.go [hunk 2]
```diff
@@ -23,46 +23,68 @@ import (
...
-	if len(scope) < 1 {
-		return root
+type rootLogger struct {
+	done uint32
+	m    sync.Mutex
+	l    *Logger
+}
+
+func (rl *rootLogger) verify() {
+	if atomic.LoadUint32(&root.done) == 0 {
```
[the remark this answers is not in the document]
Comment (hanahmily) [reply pkg/logger/setting.go:41]: `sync.Once` can't help us here. We need this logging system to provide a default logger before the user sets it with custom settings.  I got some ideas from `sync.Once` to implement:

- Return the default logger before the root logger is initialized.
- Return the root logger once it's get set.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**